### PR TITLE
199: restart

### PR DESCRIPTION
**Pull request by 0ersted**
The PR mainly deals with wallet restart

### CODE REVIEW

File: test/wallet/test_wallet.cpp [hunk 1]
```diff
@@ -229,14 +237,61 @@ TEST_F(TestWallet, normal_workflow) {
...
+
+    WALLET.reset(new Wallet("test_wallet_data//data/", 0));
+    DAG->RegisterOnLvsConfirmedCallback(
+        [&](auto vec, auto map1, auto map2) { WALLET->OnLvsConfirmed(vec, map1, map2); });
+    WALLET->CheckPassphrase("");
+    WALLET->Start();
+
+    ASSERT_EQ(balance, WALLET->GetBalance());
+
+    MINER->Start();
```
Comment (qdddddd) [inline test/wallet/test_wallet.cpp]: This line is not necessary and can be removed if `MINER->Run()` is called.

File: src/consensus/dag_manager.cpp [hunk 2]
```diff
@@ -625,7 +627,9 @@ void DAGManager::FlushToSTORE(MilestonePtr ms) {
 
         TXOC txocToRemove{std::move(utxoCreated), std::move(utxoToRemove)};
 
-        verifyThread_.Execute([=, vtxHashes = std::move(vtxHashes), txocToRemove = std::move(txocToRemove)]() {
+        verifyThread_.Execute([=, msHash = std::move(ms.cblock->GetHash()), vtxHashes = std::move(vtxHashes),
```
Comment (bassland) [inline src/consensus/dag_manager.cpp]: I think the block hash should not be moved

File: src/rpc/rpc_server.cpp [hunk 3]
```diff
@@ -191,6 +200,7 @@ grpc::Status CommanderRPCServiceImpl::SetPassphrase(grpc::ServerContext* context
         reply->set_responseinfo("Fail to set passphrase");
     } else {
         reply->set_responseinfo("Your passphrase has been successfully set!");
+        WALLET->RPCLogin();
```
Comment (bassland) [inline src/rpc/rpc_server.cpp]: I don't suggest auto login after set passphrase

File: src/rpc/rpc_server.cpp [hunk 4]
```diff
@@ -207,6 +217,7 @@ grpc::Status CommanderRPCServiceImpl::ChangePassphrase(grpc::ServerContext* cont
         reply->set_responseinfo("Fail to change passphrase. Please check passphrase");
     } else {
         reply->set_responseinfo("Your passphrase is successfully updated!");
+        WALLET->RPCLogin();
```
Comment (bassland) [inline src/rpc/rpc_server.cpp]: I don't suggest auto login after change passphrase

**Pull request merged**